pkgs/applications/editors/neovim/utils.nix

```diff
@@ -13,113 +13,45 @@
 let
   inherit (vimUtils) toVimPlugin;
 
-   /* returns everything needed for the caller to wrap its own neovim:
-   - the generated content of the future init.vim
-   - the arguments to wrap neovim with
-   The caller is responsible for writing the init.vim and adding it to the wrapped
-   arguments (["-u" writeText "init.vim" GENERATEDRC)]).
-   This makes it possible to write the config anywhere: on a per-project basis
-   .nvimrc or in $XDG_CONFIG_HOME/nvim/init.vim to avoid sideeffects.
-   Indeed, note that wrapping with `-u init.vim` has sideeffects like .nvimrc wont be loaded
-   anymore, $MYVIMRC wont be set etc
-   */
-  makeNeovimConfig =
-    { withPython3 ? true
-    /* the function you would have passed to python3.withPackages */
-    , extraPython3Packages ? (_: [ ])
-    , withNodeJs ? false
-    , withRuby ? true
-    /* the function you would have passed to lua.withPackages */
-    , extraLuaPackages ? (_: [ ])
-
-    # expects a list of plugin configuration
-    # expects { plugin=far-vim; config = "let g:far#source='rg'"; optional = false; }
-    , plugins ? []
-    # custom viml config appended after plugin-specific config
-    , customRC ? ""
-
-    # for forward compability, when adding new environments, haskell etc.
-    , ...
-    }@args:
-    let
-      rubyEnv = bundlerEnv {
-        name = "neovim-ruby-env";
-        gemdir = ./ruby_provider;
-        postBuild = ''
-          ln -sf ${ruby}/bin/* $out/bin
-        '';
-      };
-
-      # transform all plugins into an attrset
-      # { optional = bool; plugin = package; }
-      pluginsNormalized = let
+  /* transform all plugins into an attrset
+   { optional = bool; plugin = package; }
+  */
+  normalizePlugins = plugins:
+      let
         defaultPlugin = {
           plugin = null;
           config = null;
           optional = false;
         };
       in
+        # TODO use (coercedTo package (v: { plugin = v; }) pluginWithConfigType);
         map (x: defaultPlugin // (if (x ? plugin) then x else { plugin = x; })) plugins;
```

Comment:
Q: is there a module option somewhere that relates to this? I'm struggling to see how `types.coercedTo` is relevant otherwise

Reply:
I have some local code that uses nixosModule to generate the neovim config. Sadly I cant ever seem to finish it. It's not releveant here I will remove the comment